src/main/java/io/fabric8/maven/docker/service/RunService.java

```diff
@@ -412,32 +413,20 @@ private void shutdown(DockerAccess access,
                 log.error("%s", e.getMessage());
             }
         }
-        // Stop the container
-        int killGracePeriod = descriptor.getKillGracePeriod();
-        int killGracePeriodInSeconds = (killGracePeriod + 500) / 1000;
-        if (killGracePeriod != 0 && killGracePeriodInSeconds == 0) {
-            log.warn("A kill grace period of %d ms leads to no wait at all since its rounded to seconds. " +
-                     "Please use at least 500 as value for wait.kill",killGracePeriod);
-        }
-        access.stopContainer(containerId, killGracePeriodInSeconds);
-        if (killGracePeriod > 0) {
```

Comment:
There will be only a wait if killGracePeriod is > 0. I think we should do the same with the new logic, which means when no wait specified we shouldn't wait at all.